pkg/k8s/watchers/cilium_envoy_config.go

```diff
@@ -0,0 +1,316 @@
+				if cec := k8s.ObjToCEC(obj); cec != nil {
+					valid = true
+					err := k.addCiliumEnvoyConfig(cec)
+					k.K8sEventProcessed(metricCEC, metricCreate, err == nil)
+				}
+			},
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				var valid, equal bool
+				defer func() { k.K8sEventReceived(metricCEC, metricUpdate, valid, equal) }()
+
+				if oldCEC := k8s.ObjToCEC(oldObj); oldCEC != nil {
+					if newCEC := k8s.ObjToCEC(newObj); newCEC != nil {
+						valid = true
+						if newCEC.DeepEqual(oldCEC) {
+							equal = true
+							return
+						}
+						err := k.updateCiliumEnvoyConfig(oldCEC, newCEC)
+						k.K8sEventProcessed(metricCEC, metricUpdate, err == nil)
+					}
+				}
+			},
+			DeleteFunc: func(obj interface{}) {
+				var valid, equal bool
+				defer func() { k.K8sEventReceived(metricCEC, metricDelete, valid, equal) }()
+				cec := k8s.ObjToCEC(obj)
+				if cec == nil {
+					return
+				}
+				valid = true
+				err := k.deleteCiliumEnvoyConfig(cec)
+				k.K8sEventProcessed(metricCEC, metricDelete, err == nil)
+			},
+		},
+		k8s.ConvertToCiliumEnvoyConfig,
+		cecStore,
+	)
+
+	k.blockWaitGroupToSyncResources(
+		wait.NeverStop,
+		nil,
+		cecController.HasSynced,
+		k8sAPIGroupCiliumEnvoyConfigV2Alpha1,
+	)
+
+	go cecController.Run(wait.NeverStop)
+	k.k8sAPIGroups.AddAPI(k8sAPIGroupCiliumEnvoyConfigV2Alpha1)
+}
+
+func (k *K8sWatcher) addCiliumEnvoyConfig(cec *cilium_v2alpha1.CiliumEnvoyConfig) error {
+	scopedLog := log.WithFields(logrus.Fields{
+		logfields.CiliumEnvoyConfigName: cec.ObjectMeta.Name,
+		logfields.K8sNamespace:          cec.ObjectMeta.Namespace,
+		logfields.K8sUID:                cec.ObjectMeta.UID,
+		logfields.K8sAPIVersion:         cec.TypeMeta.APIVersion,
+	})
+
+	resources, err := envoy.ParseResources(cec.ObjectMeta.Namespace, cec.Spec.Resources, true, k.envoyConfigManager)
+	if err != nil {
+		scopedLog.WithError(err).Warn("Failed to add CiliumEnvoyConfig: malformed Envoy config")
+		return err
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), option.Config.EnvoyConfigTimeout)
+	defer cancel()
+	if err := k.envoyConfigManager.UpsertEnvoyResources(ctx, resources, k.envoyConfigManager); err != nil {
+		scopedLog.WithError(err).Warn("Failed to add CiliumEnvoyConfig")
+		return err
+	}
+
+	name := service.Name{Name: cec.ObjectMeta.Name, Namespace: cec.ObjectMeta.Namespace}
+	if err := k.addK8sServiceRedirects(name, &cec.Spec, resources); err != nil {
+		scopedLog.WithError(err).Warn("Failed to redirect K8s services to Envoy")
+		return err
+	}
+
+	scopedLog.Debug("Added CiliumEnvoyConfig")
+	return err
+}
+
+func getServiceName(resourceName service.Name, name, namespace string) service.Name {
+	// Default service namespace to resource namespace, if available, or "default" otherwise
+	if namespace == "" {
+		if resourceName.Namespace != "" {
+			namespace = resourceName.Namespace
+		} else {
+			namespace = "default"
+		}
+	}
+	return service.Name{Name: name, Namespace: namespace}
+}
```

Comment:
Is this the behavior that we want? My understanding is that users can specify any namespace in the `svc.Namespace` which can be a different namespace than `resourceName.Namespace`? Shouldn't they be the same? For example, in CNP we forcefully set the namespace in `fromEndpoints` to be the same namespace where the CNP is deployed even if the user manually sets a namespace different than the one where CNP is deployed.

Reply:
@aanm You tell me, I can tie this down like you suggested, clusterwide variant can then be used for mixed namespace cases if so desired.

Reply:
I think it would be better to have the same behavior as we have in CNP. Sorry I just realized I made a mistake. The `endpointSelector` of a CNP is the one that is forcefully set to the same namespace where the CNP is set but the `fromEndpoints` can allow from any namespace. It's defined [here](https://github.com/cilium/cilium/blob/314ca7baeff4f568ffc0bad95124e40665b1f88c/pkg/k8s/apis/cilium.io/utils/utils.go#L298-L316)

Reply:
So we keep as then?

Reply:
We should have the same behavior as the in the CNPs i.e., we can select from any namespace but we can only _enforce_ ingress on the namespace where cilium envoy config is.

Reply:
For Cilium Envoy Configs all Envoy listeners are on egress. I take the above to read that the redirected services MUST be in the same namespace as the CRD itself, so that service access in other namespaces is not affected. Services actually accessed by the L7 LB can be in any namespaces (unless blocked by a (Cilium) Network Policy. Sound about right?